12/homework.py

```diff
@@ -0,0 +1,77 @@
+from datetime import datetime
+import json
+
+
+# TASK 2: Create decorator with saving text into a txt file
+def my_decorator(func):
+    def inner(*args, **kwargs):
+        text = f'Function name is {func.__name__}, time of execution is {datetime.now()}\n'
+        try:
+            with open('my_deco.txt'):
+                with open('my_deco.txt', 'a') as file:
+                    file.write(text)
+        except IOError:
+            with open('my_deco.txt', 'a') as file:
+                file.write(text)
+        result = func(*args, **kwargs)
+        return result
+    return inner
+
+
+# TASK 3: Create custom exception with saving text into a txt file
+class MyCustomException(Exception):
+    def __init__(self, message='Custom exception is occurred'):
+        text = f'{message}, date of occurrence is {datetime.now()}\n'
+        try:
+            with open('my_exception.txt'):
+                with open('my_exception.txt', 'a') as file:
+                    file.write(text)
+        except IOError:
+            with open('my_exception.txt', 'w') as file:
+                file.write(text)
+        super().__init__(message)
+
+
+# TASK 1: Create saving phone book into a json file
+try:
+    with open('phone_book.json'):
```

Comment:
Why do you open this file twice here?

Reply:
For all the tasks I thought that with first "open" I would check if the file already exist and then open it in appropriate mode.
But now I understood that It works without first "open" as well. Shoud I make these changes (delete these lines of code)?
Also I see that I made mistake in line 14 - didn't change mode "a" into "w".

Reply:
Yes, remove extra `open`, please. You can use `w+`, `a+` to create a file if it does not exist before writing